flink-runtime/src/main/java/org/apache/flink/runtime/scheduler/SchedulerBase.java

```diff
@@ -187,7 +186,7 @@ public SchedulerBase(
 
 		this.executionGraph = createAndRestoreExecutionGraph(jobManagerJobMetricGroup, checkNotNull(shuffleMaster), checkNotNull(partitionTracker));
 		this.schedulingTopology = new ExecutionGraphToSchedulingTopologyAdapter(executionGraph);
-		this.failoverTopology = new DefaultFailoverTopology(executionGraph);
+		this.failoverTopology = new ExecutionGraphToSchedulingTopologyAdapter(executionGraph);
```

Comment:
Can't we use the same instance for `schedulingTopology` and `failoverTopology`?

Reply:
Sure we can. But I'd like to do it in a dedicated task #9954. And keep this PR pure refactoring work. WDYT?

Reply:
Ok, this makes sense.